aggregation/src/main/java/fk/prof/aggregation/model/FinalizedAggregationWindow.java

```diff
@@ -59,7 +64,7 @@ protected Header buildHeaderProto(int version, WorkType workType) {
     return Header.newBuilder()
         .setFormatVersion(version)
         .setWorkType(workType)
-        .setAggregationEndTime(endWithTolerance.atOffset(ZoneOffset.UTC).format(DateTimeFormatter.ISO_ZONED_DATE_TIME))
+        .setAggregationEndTime(endedAt == null ? null : endedAt.atOffset(ZoneOffset.UTC).format(DateTimeFormatter.ISO_ZONED_DATE_TIME))
```

Comment:
You mean reusing the builder object right? Also, clearing the same builder object can work only where concurrent access cannot happen. Above is a candidate, agreed